p2p/net/swarm/swarm_addr_test.go

```diff
@@ -57,3 +67,34 @@ func TestAddressesWithoutListening(t *testing.T) {
 	require.NoError(t, err)
 	require.Empty(t, a1, "expected to be listening on no addresses")
 }
+
+func TestDialAddressSelection(t *testing.T) {
```

Comment:
I'm slightly concerned about the amount of dependencies that this test adds. Not that much of a problem, now that we have a mono-repo, but maybe this would be nicer if we used a mock `transport.Transport` interface?